kernel/src/drivers/ps2/mod.rs

```diff
@@ -90,167 +93,159 @@ impl Ps2Controller {
             println!("ps2c: detected no available devices");
         }
 
-        flush_output();
+        flush_output()?;
 
         self.initialized = true;
+
+        Ok(())
     }
 
     /// Initializes the config for this controller
-    fn initialize_config(&mut self) {
+    fn initialize_config(&mut self) -> Result<(), Ps2Error> {
         // Read the config from the controller
-        self.read_config();
+        self.read_config()?;
 
         // Set all required config flags
-        self.config.set(ControllerConfigBit::PortInterrupt1, false);
-        self.config.set(ControllerConfigBit::PortInterrupt2, false);
-        self.config.set(ControllerConfigBit::PortTranslation1, false);
+        self.config.set(Ps2ConfigFlags::PORT_INTERRUPT_1, false);
+        self.config.set(Ps2ConfigFlags::PORT_INTERRUPT_2, false);
+        self.config.set(Ps2ConfigFlags::PORT_TRANSLATION_1, false);
 
         // Write the updated config back to the controller
-        self.write_config();
+        self.write_config()?;
 
         println!("ps2c: initialized config");
+
+        Ok(())
     }
 
     /// Tests this controller
-    fn test_controller(&mut self) -> bool {
-        command_ret(ControllerReturnCommand::TestController) == Some(0x55)
+    fn test_controller(&mut self) -> Result<bool, Ps2Error> {
+        Ok(command_ret(ControllerReturnCommand::TestController)? == 0x55)
     }
 
     /// Writes the current config to the PS2 controller
-    pub fn write_config(&mut self) {
-        command_data(ControllerCommand::WriteConfig, self.config.data);
+    pub fn write_config(&mut self) -> Result<(), Ps2Error> {
+        command_data(ControllerCommand::WriteConfig, self.config.bits())
     }
 
     /// Reads the config from the PS2 controller
-    pub fn read_config(&mut self) {
-        self.config.data = command_ret(ControllerReturnCommand::ReadConfig).unwrap_or(self.config.data);
-    }
-}
+    pub fn read_config(&mut self) -> Result<(), Ps2Error> {
+        let read = command_ret(ControllerReturnCommand::ReadConfig)?;
+        self.config = Ps2ConfigFlags::from_bits_truncate(read);
 
-/// Returns if the given code is OK
-#[allow(dead_code)] // To be used by keyboard / mouse driver depending on PS/2
-pub fn is_ok(code: Option<u8>) -> bool {
-    code == Some(ACK) || code == Some(SELF_TEST_PASSED)
+        Ok(())
+    }
 }
 
 /// Represents a PS2 device
 pub struct Ps2Device {
```

Comment:
Maybe this could just be Device, and Ps2Controller just be Controller, etc?

Reply:
Works better. Not used to the naming within Rust where the module names are relevant.